portal/src/graphql/portal/WebhookConfigurationScreen.tsx

```diff
@@ -38,13 +43,49 @@ interface FormState {
   timeout: number;
   totalTimeout: number;
   blocking_handlers: BlockingEventHandler[];
+  non_blocking_handler_endpoint: string;
+}
+
+function constructNonBlockingHandlerEndpoint(
+  config: PortalAPIAppConfig
+): string {
+  const result = config.hook?.non_blocking_handlers?.find(
+    (ele) => ele.events.length === 1 && ele.events[0] === "*"
+  );
+  if (result) {
+    return result.url;
+  }
+
+  return "";
+}
+
+function constructNonBlockingHandlersConfig(
+  config: PortalAPIAppConfig,
+  currentState: FormState
+): NonBlockingHookHandlerConfig[] | undefined {
+  let nonBlockingConfig = config.hook?.non_blocking_handlers ?? [];
+  // remove all "*" event handlers
+  nonBlockingConfig = nonBlockingConfig.filter(
+    (ele) => !(ele.events.length === 1 && ele.events[0] === "*")
+  );
+  // always adding non_blocking_handler in index 0
+  // to fix the json pointer for rendering error
+  if (currentState.non_blocking_handler_endpoint !== "") {
+    nonBlockingConfig.unshift({
+      events: ["*"],
+      url: currentState.non_blocking_handler_endpoint,
+    });
+  }
+
+  return nonBlockingConfig.length > 0 ? nonBlockingConfig : undefined;
```

Comment:
I think we should allow the developer to add multiple URLs, so the UI should look like the blocking ones, without event type dropdown.

Reply:
Updated!